### PR TITLE
Make use of TLS in emscripten_current_thread_process_queued_calls

### PR DESCRIPTION
**Pull request by sbc100**
The new import added to minimal_Oz_USE_PTHREADS_PROXY_TO_PTHREAD
is for `__cxa_thread_atexit`.

Fixes: #14116

### CODE REVIEW

File: tests/pthread/test_pthread_nested_work_queue.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include <stdio.h>
+#include <emscripten/threading.h>
+#include <emscripten/emscripten.h>
+
+void work() {
+  // This `work` item is called form within
```
Comment (kripken) [inline tests/pthread/test_pthread_nested_work_queue.c:6]:
```suggestion
  // This `work` item is called from within
```

**Review by kripken — APPROVED** [review]
Lgtm but definitely @juj should weigh in here.

**Comment by sbc100** [comment]
ping @juj

**Comment by sbc100** [comment]
Maybe we can land and get @juj's feedback after?   I think this is pretty safe, especially given that it includes a test case that fails before and passes after this change.

**Comment by kripken** [comment]
sgtm to land

**Pull request merged**